File: Data Structures & Algorithms/majority-element-ii/submission-0.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int target = nums.length/3;
        Map<Integer, Integer> freq = new HashMap();
        for(int n : nums){
            freq.put(n, freq.getOrDefault(n, 0) + 1);
        }
        List<Integer> ans = new ArrayList<>();
        for(Map.Entry<Integer, Integer> entry : freq.entrySet()){
            if(entry.getValue() > target){
                ans.add(entry.getKey());
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/minimum-recolors-to-get-k-consecutive-black-blocks/submission-0.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        freq = {}
        freq['W'] = 0
        freq['B'] = 0

        for i in range(k):
            freq[blocks[i]] += 1
        
        l = 0
        r = k - 1

        ans = freq['W']

        while r+1 < len(blocks):
            freq[blocks[r+1]] += 1
            freq[blocks[l]] -= 1
            l += 1
            r += 1

            ans = min(ans, freq['W'])
        return ans
        